server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -1010,62 +1031,97 @@ public void changeOffset(String path, long offset) throws MetadataException {
     }
   }
 
+  public void changeAlias(String path, String alias) throws MetadataException {
+    lock.writeLock().lock();
+    try {
+      LeafMNode leafMNode = (LeafMNode) mtree.getNodeByPath(path);
+      leafMNode.getParent().deleteAliasChild(leafMNode.getAlias());
+      leafMNode.getParent().addAlias(alias, leafMNode);
+      leafMNode.setAlias(alias);
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
   /**
    * upsert tags and attributes key-value for the timeseries if the key has existed, just use the
    * new value to update it.
    *
+   * @param alias         newly added alias
    * @param tagsMap       newly added tags map
    * @param attributesMap newly added attributes map
    * @param fullPath      timeseries
    */
-  public void upsertTagsAndAttributes(
-      Map<String, String> tagsMap, Map<String, String> attributesMap, String fullPath)
-      throws MetadataException, IOException {
+  public void upsertTagsAndAttributes(String alias, Map<String, String> tagsMap,
+      Map<String, String> attributesMap, String fullPath) throws MetadataException, IOException {
     lock.writeLock().lock();
     try {
       MNode mNode = mtree.getNodeByPath(fullPath);
       if (!(mNode instanceof LeafMNode)) {
         throw new PathNotExistException(fullPath);
       }
       LeafMNode leafMNode = (LeafMNode) mNode;
+      // upsert alias
+      if (alias != null) {
+        if (leafMNode.getParent().hasChild(alias)) {
+          throw new MetadataException("The alias already exits.");
```

Comment:
I think, if the alias exists, just keep it and do the next instead of throwing an exception is better. What do you think? And a typo: exist

Reply:
Yes!

Reply:
fix